### ISSUE TITLE
Detect  whether there is an uncommitted files judgment error in the repository when use git_merge

### ISSUE THREAD

**Issue by xiaofuhub** [libgit2/libgit2#5630]
You are opening a _bug report_ against the libgit2 project: we use
GitHub Issues for tracking bug reports and feature requests.  If you
have a question about an API or usage, please ask on StackOverflow:
http://stackoverflow.com/questions/tagged/libgit2.  If you want to
have high-level discussions about the libgit2 project itself, visit
https://github.com/libgit2/discussions.

Otherwise, to report a bug, please fill out the reproduction steps
(below) and delete these introductory paragraphs.  Thanks!

**Comment by xiaofuhub** [comment]
I 'm trying to pull from a repo with libgit2.

My steps are these:

I use git_remote_fetch to down remote origin data and the return OK.

after I use git_merge API.

The question is when I just deleted a file 'aa.txt' in local branch 'master', But not commit it. At the same times I Merge remote branch('origin/master') head. the remote head just modify 'bb.txt'. But when I use git_merge it return error code '-13'. the error info is "1 uncommitted change would be overwritten by merge". I just deleted a file in my local branch.

But I can successed to execute in Git command line 'git pull'.

so, I suspect that my strategy is wrong when I execute Checkout. How to avoid this error？

Stackoverflow question location: [](https://stackoverflow.com/questions/63546380/how-to-set-checkout-strategy-in-libgit2)
